### PR TITLE
Test that plugins can handle non-ascii characters in paths.

### PR DESCRIPTION
**Pull request by pelmers**
Add a test with all plugins enabled that contains CJK non-ascii (but still unicode) characters in paths, to extend the previous non-utf8 path handling test.

### CODE REVIEW

File: tests/test_non_ascii_paths/code/makefile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+all: c
```
Comment (erikrose) [inline tests/test_non_ascii_paths/code/makefile]: (Commenting here because it won't let me comment on a binary file.) Did you mean to check "c" in?

File: tests/test_non_ascii_paths/code/makefile [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+	mkdir code
+	cd code && python -c "import os; os.mkdir('\x8d\xae\xa2\xa0\xef')"
+	cd code && python -c "open('\x8d\xae\xa2\xa0\xef/\xaf\xa0\xaf\xaa\xa0', 'w').write('Hello!\n')"
+c: 第一.c
```
Comment (erikrose) [inline tests/test_non_ascii_paths/code/makefile]: I bet we're also going to test various versions of `make` pretty effectively. ;-)

File: tests/test_non_ascii_paths/code/makefile (hunk 1 above)
Comment (pelmers) [inline tests/test_non_ascii_paths/code/makefile]: Nope, thanks for the catch; I forgot to add it on the clean command.

Comment (erikrose) [inline tests/test_non_ascii_paths/code/makefile]: And stick it in gitignore if git doesn't already ignore it.

**Comment by erikrose** [comment]
So you've turned the non-UTF8 tests into merely non-ASCII ones. While this is nice for HFS+ people like myself, do we lose anything important?

**Comment by pelmers** [comment]
We keep the non-UTF8 tests (non-utf8 is a subset of non-ascii)

**Comment by erikrose** [comment]
sgtm! r+ when the tests pass.
